v2/logging.go

```diff
@@ -1,45 +1,23 @@
 package shuttle
 
 import (
-	"context"
-	"fmt"
-	"os"
-	"time"
+	"io"
+	"log/slog"
 )
 
-type Logger interface {
```

Comment:
since the interface that we expose and use for logging has Info, Warn and Error, can we keep it for now, and hook the slog specific func behind it as default?

Reply:
added back Logger interface